.travis.yml

```diff
@@ -1,66 +1,34 @@
 dist: xenial
-sudo: required
+os: linux
 language: cpp
 
-services:
-  - xvfb
-
-matrix:
-  include:
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-6
-      env: COMPILER=gcc GCC=6
-    - os: linux
-      if: branch = master
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-7
-      env: COMPILER=gcc GCC=7
+stages:
+  - test
+  - singularity-build-push
 
 env:
   global:
     - DISPLAY=":99.0"
-    - MINCONDA_VERSION="latest"
-    - MINCONDA_LINUX="Linux-x86_64"
+  jobs:
+    - PYTHON_VERSION="3.7.5"
+    - PYTHON_VERSION="3.6.9"
 
+services:
+  - xvfb
 before_install:
-  - |
-    if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-      if [[ "$COMPILER" == "gcc" ]]; then
```

Comment:
I have this `if` because I also want to test compilers like `Clang` here. Nevertheless, the operational environment has `g++6` and `centos`. My first thought is that we may not want to put all the tests into Singularity because it is only meant for our operational environment. 

So the question is, can we separate them in `Travis`? i.e. a widely covered non-singularity test and a singularity test. Another immediate solution is to move former to the `Azure` pipeline.

Reply:
Currently the test stage and the singularity image building stages are decoupled almost completely, the only relation between the two is that if the tests fail then the image won't build.

So the tests could be expanded on even more (e.g. additional compilers) without affecting the image build process, if that's what you mean.

Reply:
Yes! I would like to expand the test stage! See my comment above.